packages/@okta/vuepress-site/use_cases/inline_hooks/index.md

```diff
@@ -121,7 +122,37 @@ The names of commands follow Java-style reverse DNS name format, beginning with
+| errorCauses  | An array of ErrorCause objects.         | Array of ErrorCauses |
+
+The `errorSummary` should be a general statement of any problem the external service encountered in handling the request from Okta. The `errorCauses` are intended to provide more detailed information and are particularly helpful if there were multiple problems.
+
+An `ErrorCause` object must include the following fields:
+
+| Property     | Description                                                                                                                                                         | Data Type |
+|--------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
+| errorSummary | Human-readable summary of the error.                                                                                                                                | String    |
+| reason       | A brief, enum-like string indicating the nature of the error, e.g., `UNIQUE_CONSTRAINT` for a property uniqueness violation.                                        | String    |
+| locationType | Where in the request the error was found (`body`, `header`, `url`, or `query`).                                                                                     | String    |
+| location     | The valid JSON path to the location of the error. E.g., if there was an error in the user's `login` field, the `location` should be `data.userProfile.login`.       | String    |
+| domain       | Indicates the source of the error. If the error was on the user's profile, use `end-user`. If the error resulted from the external service, use `external-service`. | String    |
+
+While there are no technical restrictions on the values for any of the fields in an `ErrorCause` object, using them as described in the table above allows you to provide rich error information that, along with the `debugContext`, can be very useful in determining why a user's registration failed.
+
+### debugContext
+
+The `debugContext` object is a free-form JSON object where you can provide information to be logged in the `debugContext` field of `inline_hook.response.processed` events in the Okta System Log. These events are fired when Okta has received a response from the external service but there is either:
+
+- An error object in the response body (provided by the external service)
+- A problem with the response body, for example, inclusion of an invalid command. 
+
+The `debugContext` object lets the external service provide information about its handling of the hook execution request (e.g., execution time, information that validated, etc.). This can be helpful for diagnosing what went wrong when a registration fails.
+
+> Note: `inline_hook.response.processed` events are _not_ fired following a successful registration unless debugging is enabled for the configured hook.
```

Comment:
I'd suggest re-wording to something like - `Note: inline_hook.response.processed events are fired only in case of failure in processing the response from your service used in the inline hook.`.
As Will said, the `debug` mode is not public information.

Reply:
For the other point around syslog consistency - I think we should include a note that says something along the lines of `Currently, this event is only available for inline hooks of type com.okta.user.pre-registration`.
/cc @ankurdatta-okta

Reply:
I think if we're not going to mention `debug` mode, this note should just be omitted entirely; we already describe the scenarios in which the event is fired a few paragraphs up. 

To your second point, @amoghvasekar-okta, it seems this event is also used by import hooks, so the note should include them as well.

Reply:
**Regarding including info on Syslog events:**
We have a few pending tasks to include more documentation on troubleshooting of both Inline and Event Hooks. This will include more info on corresponding Syslog events (e.g. See this note: https://github.com/okta/okta-developer-docs/pull/168#discussion_r272767888 and this JIRA: https://oktainc.atlassian.net/browse/OKTA-222651)

So for the time being, it would be better to not mention anything related to the Inline Hook Syslog events until we add the section on troubleshooting in our docs. We have briefly discussed this with @nbarbettini and will make necessary updates in the upcoming sprints for the InfoDev team.

**Regarding info on debug mode:**
Since this info is not public yet, let us not mention the debug mode anywhere either.

Reply:
So @ankurdatta-okta, if you don't want to talk about Syslog events, it seems like there's no sense in mentioning the `debugContext` at all. In that case, we should make sure that it's omitted from the JSON examples as well.

Reply:
Removed the section on debugContext from the inline hooks overview page, and removed the debugContext object from the sample response listing in the registration inline hook page.